lib/backend/grpcclient/grpc.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcclient
+
+import (
+	"context"
+	"time"
+
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/trace"
+	"github.com/jonboulle/clockwork"
+
+	log "github.com/sirupsen/logrus"
+)
+
+type GrpcBackend struct {
+	clock clockwork.Clock
+}
+
+// Config represents JSON config for grpc backend
+type Config struct {
+	// TODO(michaelmcallister): implementation
```

Comment:
What's your plan for the actual implementation here?

Reply:
The following configuration will be needed at the very minimum:

- gRPC server address
- serverCA
- ClientCert
- ClientKey

I can add this in now if you'd prefer, I don't mind